staging/src/k8s.io/apimachinery/pkg/util/wait/wait_test.go

```diff
@@ -441,6 +442,8 @@ func TestPollForever(t *testing.T) {
 			if !open {
 				t.Fatalf("did not expect channel to be closed")
 			}
+		case err := <-errc:
+			t.Fatal(err)
```

Comment:
I think this check should be moved to after line 463, because until PollInfinite is possible to send err to errc until it returns. After line 463, we know that PollInifite has returned, so checking errc there makes sure that PollInfinite never reports error.

Reply:
I think you're right and at least if `PollInfinite` returns an error after the 10th invocation of `f` the error will never be reported. I think we need to handle it here as well, though, to make sure a potential error within the first ten invocations of `f` is being reported. Otherwise `t.Fatalf` in line 443 would be invoked only.
Applied you feedback in e480d730ff9b5eae00aca0b894d3ce1495ea040b

Reply:
I agree that in case `ch` is closed, we want to report the `err` from `<-errc` as well.

But with the current implementation, only one of `<-errc` case and `<-ch` case will be reported. How about something like this:

```golang
for i := 0; i < 10; i++ {
  select {
  case _, open := <-ch:
     if !open {
       if len(errc) != 0 {
          t.Fatalf("did not expect channel to be closed, unexpected err: %v", <-errc)
       }
       t.Fatalf("did not expect channel to be closed")
     }
  case <-time.After(ForeverTestTimeout):
    t.Fatalf("channel did not return at least once within the poll interval")
  }
}
```

Reply:
Really like your approach, have incorporated it. Thanks a lot!